src/app/shared/services/dynamic-data/adapters/persistedMemory.ts

```diff
@@ -138,15 +138,35 @@ export class PersistedMemoryAdapter {
       }
     }
     await this.collection.bulkUpsert(updates);
+
+    // Handle deletions (check for flows that exist in DB but not in state, and remove from DB)
+    const db = await this.mapDBToObject({});
+    const idsToDelete = [];
+    for (const flow_type of Object.keys(db)) {
+      const { deleted } = compareObjectKeys(db[flow_type], this.state[flow_type]);
```

Comment:
@jfmcquade 
Could you confirm I've got this the right way round? The previous logic confused me a little